api/historydata.php

```
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: joshua
+ * Date: 9/27/2017
+ * Time: 1:04 PM
+ */
+
+header('Content-Type: text/plain; charset=utf-8');
+
+$allowed_platforms = array('android', 'ios', 'mac', 'web', 'windows', 'developer');
+$allowed_versions = array('1.0'); // api versions, not product versions
+//$release_tiers = array('alpha', 'beta', 'stable');
+
+function fail($s) {
+  header("HTTP/1.0 400 $s");
+  exit;
+}
+
+// Mirroring versioninfo.php parameters.
+
+if(isset($_REQUEST['version'])) {
+  $version = $_REQUEST['version'];
+} else {
+  $version = '1.0';
+}
+
+if(array_search($version, $allowed_versions) === FALSE) {
+  fail('Invalid version: Only '.implode('/', $allowed_versions).' allowed');
+}
+
+if(isset($_REQUEST['platform'])) {
+  $platform = $_REQUEST['platform'];
+  if(!preg_match('/^('.implode('|', $allowed_platforms).')$/', $platform)) {
+    fail('Invalid platform: Only '.implode('/', $allowed_platforms).' allowed');
+  }
+  $platforms = array($platform);
+} else {
+  $platforms = $allowed_platforms;
+}
+
+// Doing the actual work.
+
+function get_history_contents($platform) {
+// Reads straight from a file.  Likely to be useful for the history-exposing API.
+  $inputURL = "../$platform/history.md";
+
+  $contents = @file_get_contents($inputURL);
+
+  if($contents == FALSE) {
+    fail("Cannot locate history information!");
+  }
+  echo $contents;
+}
+
+for($i=0; $i < count($platforms); $i++) {
```

I don't see a real need to support dumping history from all platforms at once. One at a time is fine.

---

So, it should return an error if a platform isn't specified, then?  I figured it should probably have *some* kind of default behavior, like with versioninfo.php, though this default does seem admittedly less useful.

---

Yes, just return an error if the platform isn't specified. It's an internal API so we don't have to be too friendly in error cases.